File: src/main/java/com/mohe/spring/repository/PlaceDescriptionVectorRepository.java
package com.mohe.spring.repository;

import com.mohe.spring.entity.PlaceDescriptionVector;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface PlaceDescriptionVectorRepository extends JpaRepository<PlaceDescriptionVector, Long> {
    
    Optional<PlaceDescriptionVector> findByPlaceId(Long placeId);
    
    void deleteByPlaceId(Long placeId);
    
    List<PlaceDescriptionVector> findByPlaceIdIn(List<Long> placeIds);
    
    @Query("SELECT pdv FROM PlaceDescriptionVector pdv WHERE pdv.place.shouldRecheckRating = false")
    List<PlaceDescriptionVector> findAllActive();
}

File: src/main/kotlin/com/mohe/spring/repository/PlaceDescriptionVectorRepository.kt
package com.mohe.spring.repository

import com.mohe.spring.entity.PlaceDescriptionVector
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface PlaceDescriptionVectorRepository : JpaRepository<PlaceDescriptionVector, Long> {
    fun findByPlaceId(placeId: Long): PlaceDescriptionVector?
    fun deleteByPlaceId(placeId: Long)
    fun findByPlaceIdIn(placeIds: List<Long>): List<PlaceDescriptionVector>
    
    @Query("SELECT pdv FROM PlaceDescriptionVector pdv WHERE pdv.place.shouldRecheckRating = false")
    fun findAllActive(): List<PlaceDescriptionVector>
}